src/library_manager/llext_manager.c

```diff
@@ -71,6 +72,32 @@ static int llext_manager_align_unmap(void __sparse_cache *vma, size_t size)
 	return sys_mm_drv_unmap_region(aligned_vma, ALIGN_UP(pre_pad_size + size, PAGE_SZ));
 }
 
+static void llext_manager_detached_map(void __sparse_cache *vma, size_t size, uint32_t flags)
+{
+#ifdef CONFIG_MMU
+	size_t pre_pad_size = (uintptr_t)vma & (PAGE_SZ - 1);
+	void *aligned_vma = (__sparse_force uint8_t *)vma - pre_pad_size;
+
+	/* Use cached virtual address */
+	uintptr_t va = POINTER_TO_UINT(sys_cache_cached_ptr_get(aligned_vma));
+
+	arch_mem_map(aligned_vma, va, ALIGN_UP(pre_pad_size + size, PAGE_SZ), flags);
```

Comment:
In principle you don't have to page-align the size, `arch_mem_map()` does it already, but doesn't hurt too much